File: content-solutions/queue-deque/queue-carousel-line.js
function solution(commands) {
  const line = [];
  let head = 0;
  const answer = [];
  for (const command of commands) {
    const [name, value] = command.split(" ");
    if (name === "arrive") {
      line.push(value);
    } else {
      const group = [];
      const count = Math.min(Number(value), line.length - head);
      for (let i = 0; i < count; i++) group.push(line[head++]);
      answer.push(group);
    }
  }
  return answer;
}

File: content-solutions/queue-deque/queue-carousel-line.java
import java.util.*;

class Solution {
    public List<List<String>> solution(String[] commands) {
        Deque<String> line = new ArrayDeque<>();
        List<List<String>> answer = new ArrayList<>();
        for (String command : commands) {
            String[] parts = command.split(" ");
            if (parts[0].equals("arrive")) {
                line.offer(parts[1]);
            } else {
                List<String> group = new ArrayList<>();
                int count = Math.min(Integer.parseInt(parts[1]), line.size());
                for (int i = 0; i < count; i++) group.add(line.poll());
                answer.add(group);
            }
        }
        return answer;
    }
}
